gamification-github-services/src/main/java/org/exoplatform/gamification/github/plugin/GithubConnectorPlugin.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * This file is part of the Meeds project (https://meeds.io/).
+ * Copyright (C) 2020 - 2023 Meeds Association contact@meeds.io
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+package org.exoplatform.gamification.github.plugin;
+
+import com.github.scribejava.apis.GitHubApi;
+import com.github.scribejava.core.builder.ServiceBuilder;
+import com.github.scribejava.core.model.OAuth2AccessToken;
+import com.github.scribejava.core.oauth.OAuth20Service;
+import io.meeds.gamification.model.RemoteConnectorSettings;
+import io.meeds.gamification.plugin.ConnectorPlugin;
+import io.meeds.gamification.service.ConnectorSettingService;
+import io.meeds.oauth.exception.OAuthException;
+import io.meeds.oauth.exception.OAuthExceptionCode;
+import org.apache.commons.lang.StringUtils;
+import org.exoplatform.commons.ObjectAlreadyExistsException;
+import org.exoplatform.commons.exception.ObjectNotFoundException;
+import org.exoplatform.gamification.github.model.GithubAccessTokenContext;
+import org.exoplatform.gamification.github.model.GithubAccount;
+import org.exoplatform.gamification.github.services.GithubAccountService;
+import org.exoplatform.services.log.ExoLogger;
+import org.exoplatform.services.log.Log;
+import org.exoplatform.services.security.Identity;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.concurrent.ExecutionException;
+
+public class GithubConnectorPlugin extends ConnectorPlugin {
+
+  private static final Log              LOG                = ExoLogger.getLogger(GithubConnectorPlugin.class);
+
+  private static final String           CONNECTOR_NAME     = "github";
+
+  private static final String           CONNECTOR_SCOPE    = "user:email";
```

Comment:
No need to read user email, just user identifier.

```suggestion
  private static final String           CONNECTOR_SCOPE    = "read:user";
```

Reply:
Done!